File: java/word_wrap/src/test/java/com/devartis/wordwrap/WordWrapTest.java
package com.devartis.wordwrap;

import org.junit.Test;

import static com.devartis.wordwrap.WordWrapper.wrap;
import static org.junit.Assert.assertEquals;


public class WordWrapTest {

    @Test
    public void noWrapOnEmptyString() {
        assertEquals("", wrap("", 1));
    }

    @Test
    public void noWrapWhenStringShorterThanLimit() {
        assertEquals("string", wrap("string", 10));
    }

    @Test
    public void wrapTwoWordsAfterSpace() {
        assertEquals("word\nword", wrap("word word", 6));
    }

    @Test
    public void wrapThreeWordsAfterFirstSpace() {
        assertEquals("word\nword\nword", wrap("word word word", 6));
    }

    @Test
    public void wrapThreeWordsAfterSecondSpace() {
        assertEquals("word word\nword", wrap("word word word", 11));
    }

    @Test
    public void splitOneWord() {
        assertEquals("wo\nrd", wrap("word", 2));
    }

    @Test
    public void splitOneWordManyTimes() {
        assertEquals("abc\ndef\nghi\nj", wrap("abcdefghij", 3));
    }

    @Test
    public void wrapBeforeWordBoundary() {
        assertEquals("wor\nd\nwor\nd", wrap("word word", 3));
    }
    
    @Test
    public void includeWordAfterBreak() {
        assertEquals("word\ns hi\nword\ns\nword\ns", wrap("words hi words words", 4));
    }
}

File: python/word_wrap/word_wrap_test.py
from nose.tools import istest, eq_

from word_wrap import wrap


@istest
def no_wrap_on_empty_string():
    eq_('', wrap('', 1))


@istest
def no_wrap_when_string_shorter_than_limit():
    eq_('string', wrap('string', 10))


@istest
def split_one_word():
    eq_('wo\nrd', wrap('word', 2))


@istest
def split_one_word_many_times():
    eq_('abc\ndef\nghi\nj', wrap('abcdefghij', 3))


@istest
def wrap_two_words_after_space():
    eq_('word\nword', wrap('word word', 6))


@istest
def wrap_three_words_after_first_space():
    eq_('word\nword\nword', wrap('word word word', 6))


@istest
def wrap_three_words_after_second_space():
    eq_('word word\nword', wrap('word word word', 11))


@istest
def wrap_before_word_boundary():
    eq_('wor\nd\nwor\nd', wrap('word word', 3))


# Nice to have.

@istest
def wrap_three_words_after_second_space():
    eq_('word\ns hi\nword\ns\nword\ns', wrap('words hi words words', 4))
